components/dropdown/dropdown-content-mixin.js

```diff
@@ -694,18 +776,24 @@ export const DropdownContentMixin = superclass => class extends RtlMixin(supercl
 
 		return html`
 			<div class="d2l-dropdown-content-position" style=${styleMap(positionStyle)}>
-				<div class="d2l-dropdown-content-width" style=${styleMap(widthStyle)}>
+				<div  id="d2l-dropdown-wrapper" class="d2l-dropdown-content-width" style=${styleMap(widthStyle)} ?closing="${this._closing}">
 					<div class=${classMap(topClasses)} style=${styleMap(contentWidthStyle)}>
 						<slot name="header" @slotchange="${this.__handleHeaderSlotChange}"></slot>
 					</div>
 					<div class="d2l-dropdown-content-container" style=${styleMap(contentStyle)} @scroll=${this.__toggleScrollStyles}>
 						<slot class="d2l-dropdown-content-slot"></slot>
 					</div>
 					<div class=${classMap(bottomClasses)} style=${styleMap(contentWidthStyle)}>
-						<slot name="footer" @slotchange="${this.__handleFooterSlotChange}"></slot>
+						<slot name="footer" @slotchange="${this.__handleFooterSlotChange}">
+						</slot>
+						<d2l-button
+						style=${styleMap(closeButtonVisibility)} 
+						@click=${this.close}>
+						${this.localize('components.dropdown.close')}</d2l-button>
 					</div>
 				</div>
 			</div>
+			<d2l-backdrop for-target="d2l-dropdown-wrapper" ?shown="${specialMobileStyle && this.opened && !this._closing}" ></d2l-backdrop>
```

Comment:
I'd be interested in avoiding rendering `d2l-backdrop` in cases where it would not be used... i.e. when there is no `specialMobileStyle`.  So, it would become something like:

```javascript
const dropdown = html`
   <div class="d2l-dropdown-content-position" style=${styleMap(positionStyle)}>
      ...
   </div>
`;

if (specialMobileStyle) return html`${dropdown} ${html`<d2l-backdrop ...></d2l-backdrop>`}`;
else return dropdown;
```